### ISSUE TITLE
Sandbox.usingPromise does not set promise library for mocks

### ISSUE THREAD

**Issue by nhinds** [sinonjs/sinon#1863]
**Describe the bug**
Calling `.usingPromise(promiseLibrary)` on a sandbox works for stubs created from the sandbox, but does not appear to work for mocks created from the sandbox. Mocks continue to use the node.js default Promise library

**To Reproduce**
Create a sandbox using e.g. Bluebird as its promise library, and attempt to create stubs/mocks which return Promises using `.resolves`:
```js
const sinon = require('sinon');

// Set up a sandbox using bluebird as the promise library
const sandbox = sinon.createSandbox().usingPromise(require('bluebird'));

// Call a stub which returns a resolved promise
const stubPromise = sandbox.stub().resolves('value for stub')();
// Call bluebird API methods on the returned Promise object
stubPromise.tap(console.log).return('final value for stub').then(console.log);
// Works :)

// Call a mock which returns a resolved promise
const mockPromise = sandbox.mock().resolves('value for mock')();
// Call bluebird API methods on the returned Promise object
mockPromise.tap(console.log).return('final value for mock').then(console.log);
// TypeError: mockPromise.tap is not a function

// Call a method on a mock which returns a resolved promise
const objectToMock = { method: function() {} };
sandbox.mock(objectToMock).expects('method').resolves('value for mock method');
const mockMethodPromise = objectToMock.method();
// Call bluebird API methods on the returned Promise object
mockMethodPromise.tap(console.log).return('final value for mock method').then(console.log);
// TypeError: mockMethodPromise.tap is not a function
```

**Expected behavior**
I would expect mocks returned from the sandbox to return Promises using the library passed to `sandbox.usingPromise()`.

As a workaround, `.usingPromise(...)` can be called on each stubbing chain on each mock, e.g.:
```js
const mockPromise = sandbox.mock().usingPromise(require('bluebird')).resolves('value for mock')();

sandbox.mock(objectToMock).expects('method').usingPromise(require('bluebird')).resolves('value for mock method');
```

**Context (please complete the following information):**

* Library version:  6.1.3
* Environment: node.js v8.10.0
* Other libraries you are using: bluebird 3.5.1

**Additional context**
N/A

**Comment by mroderick** [comment]
Thank you for your bug report.

That does indeed look like a defect.

Do you want to make a pull request to fix it?

**Comment by nhinds** [comment]
Sure, opened #1872.

**Comment by fatso83** [comment]
Merged #1862 in 6.2
